python/lsst/dax/apdb/cassandra/apdbCassandra.py

```diff
@@ -725,7 +725,36 @@ def getDiaForcedSources(
 
     def containsVisitDetector(self, visit: int, detector: int) -> bool:
         # docstring is inherited from a base class
-        raise NotImplementedError()
+        # The order of checks corresponds to order in store(), on potential
+        # store failure earlier tables have higher probability containing
+        # stored records. With per-partition tables there will be many tables
+        # in the list, but it is unlikely that we'll use that setup in
+        # production.
+        existing_tables = self._schema.existing_tables(ApdbTables.DiaSource, ApdbTables.DiaForcedSource)
+        tables_to_check = existing_tables[ApdbTables.DiaSource][:]
+        if self.config.use_insert_id:
+            tables_to_check.append(self._schema.tableName(ExtraTables.DiaSourceChunks))
```

Comment:
Do you actually need to check the insert ID/chunk tables? I was under the impression that they weren't part of the logical state of the DB and don't affect association. Or is the problem that a duplicate insert of a DiaSource could still break on that table?

Reply:
For now this is mostly for defensive reasons, but also because I want to understand timing of those operations, it may be more efficient to check chunk tables first, but it would depend on relative timing. I want to gather some metrics before trying the other option.